pke/kyber/internal/common/field_test.go

```diff
@@ -0,0 +1,86 @@
+package common
+
+import (
+	"flag"
+	"math/rand"
+	"testing"
+)
+
+var runVeryLongTest = flag.Bool("very-long", false, "runs very long tests")
+
+func modQ32(x int32) int16 {
+	y := int16(x % int32(Q))
+	if y < 0 {
+		y += Q
+	}
+	return y
+}
+
+func TestBarrettReduceFull(t *testing.T) {
+	if !*runVeryLongTest {
+		t.SkipNow()
+	}
+	for x := -1 << 15; x <= 1<<15; x++ {
+		y1 := barrettReduce(int16(x))
+		y2 := int16(x) % Q
+		if y2 < 0 {
+			y2 += Q
+		}
+		if x < 0 && int16(-x)%Q == 0 {
+			y1 -= Q
+		}
+		if y1 != y2 {
+			t.Fatalf("%d %d %d", x, y1, y2)
+		}
+	}
+}
+
+func TestMontReduce(t *testing.T) {
+	for i := 0; i < 1000; i++ {
+		x := rand.Int31n(int32(Q)*(1<<16)) - int32(Q)*(1<<15) // nolint:gosec
```

Comment:
I think they aren't needed for tests

```suggestion
		x := rand.Int31n(int32(Q)*(1<<16)) - int32(Q)*(1<<15)
```

Reply:
It did trigger for me.